### PR TITLE
[MAIN] [STRATCONN] added depends_on conditions on salesforce actions

### PR DESCRIPTION
**Pull request by AnkitSegment**
Applied depends_on condition on enableConsentMode field SALESFORCE ACTIONS

JIRA: https://segment.atlassian.net/browse/STRATCONN-3849

Stage Testing snapshot:

https://github.com/segmentio/action-destinations/assets/139338151/2427f39c-2e72-42aa-9890-b6e048ffd0d2

_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [X] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [X] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Review by nick-Ag — COMMENTED** [review]
I think for all of these fields you'd want to update the conditions to also show the fields if operation == 'update'. Otherwise we block out users from updating key fields with the update operation

**Comment by AnkitSegment** [comment]
> I think for all of these fields you'd want to update the conditions to also show the fields if operation == 'update'. Otherwise we block out users from updating key fields with the update operation

@nick-Ag yes, updated the conditions accordingly.

File: packages/destination-actions/src/destinations/salesforce/lead/index.ts [hunk 1]
```diff
@@ -50,6 +50,15 @@ const action: ActionDefinition<Settings, Payload> = {
           then: { '@path': '$.traits.last_name' },
           else: { '@path': '$.properties.last_name' }
         }
+      },
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/salesforce/lead/index.ts:53]: I read thru the code for the delete operation and we use the `traits` object field to lookup a record ID to delete. I think if a user selects operation: delete we can hide every field other than 'operation', 'recordMatcherOperator', and 'traits'. I suggest you define the conditions one time in `sf-properties.ts` like:
```
export const hideIfDeleteOperation: DependsOnConditions = {
  conditions: [
    {
      fieldKey: 'operation',
      operator: 'is_not',
      value: 'delete'
    }
  ]
}
```
And import it and use for all applicable fields. Sorry for the additional blocking comment but I think that's all this one needs

**Review by nick-Ag — DISMISSED** [review]
See comment above about hiding more fields when `operation: delete`

**Comment by nick-Ag** [comment]
Looks good for the account action 👍

**Comment by nick-Ag** [comment]
@AnkitSegment - could you please also add the conditions to the cases and customObject actions?

Cases: enable_batching, batch_size, bulkUpsertExternalId, bulkUpdateRecordId, description, customFields
customObject: same as cases, be sure not to apply to customObjectName field

Going to approve anyways, thanks for your help here

**Review by nick-Ag — APPROVED** [review]

**Review by nick-Ag — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @AnkitSegment PR deployed. It was not pushed. Let me know if it needs a push

**Pull request merged**